### PR TITLE
fix: creating IVF_HNSW index fails with cosine metric type

### PR DESCRIPTION
**Pull request by BubbleCal**
- normalize vectors by `normalize_fsl`
- change the default HNSW build params `ef_constrction: 100 -> 150`
- fix the ground truth for cosine

### CODE REVIEW

File: rust/lance/src/index/vector/ivf/io.rs [hunk 1]
```diff
@@ -446,12 +446,12 @@ async fn build_hnsw_quantization_partition(
     let mut metric_type = metric_type;
     if metric_type == MetricType::Cosine {
         // Normalize vectors for cosine similarity
-        vectors = normalize_arrow(&vectors)?;
+        vectors =
+            Arc::new(spawn_cpu(move || Ok(normalize_fsl(vectors.as_fixed_size_list())?)).await?);
```
Comment (eddyxu) [inline rust/lance/src/index/vector/ivf/io.rs:450]: How big is this `vectos` usually? is it worth to create a new thread?

File: rust/lance/src/index/vector/ivf.rs [hunk 2]
```diff
@@ -2226,15 +2231,11 @@ mod tests {
...
-        let ivf_centroids = FixedSizeListArray::try_new_from_values(centroids, DIM as i32).unwrap();
-        let ivf_params =
-            IvfBuildParams::try_with_centroids(nlist, Arc::new(ivf_centroids)).unwrap();
-
+        let ivf_params = IvfBuildParams::new(nlist);
         let pq_params = PQBuildParams::default();
         let hnsw_params = HnswBuildParams::default();
         let params = VectorIndexParams::with_ivf_hnsw_pq_params(
-            MetricType::L2,
+            MetricType::Cosine,
```
Comment (eddyxu) [inline rust/lance/src/index/vector/ivf.rs:2238]: should we create a new test for Cosine? Change it to cosine will lose converage on L2?

File: rust/lance/src/index/vector/ivf/io.rs (hunk 1 above)
Comment (BubbleCal) [reply rust/lance/src/index/vector/ivf/io.rs:450]: it's all the vectors of one partition, could be very big for large dataset

File: rust/lance/src/index/vector/ivf.rs (hunk 2 above)
Comment (BubbleCal) [reply rust/lance/src/index/vector/ivf.rs:2238]: the Cosine also covers L2 path, it's L2 + normalization

Comment (BubbleCal) [reply rust/lance/src/index/vector/ivf.rs:2238]: we can keep one of the test using L2

**Review by eddyxu — APPROVED** [review]

**Pull request merged**